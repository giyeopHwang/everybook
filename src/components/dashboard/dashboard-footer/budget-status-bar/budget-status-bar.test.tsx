import { screen } from '@testing-library/react';
import { renderWithWrapper } from '@/test/test-utils';
import BudgetStatusBar from './budget-status-bar.component';

describe('BudgetStatusBar', () => {
  it('renders BudgetStatusBar component', () => {
    const budgetStatusBar = renderWithWrapper(<BudgetStatusBar />);
    expect(budgetStatusBar).toMatchSnapshot();

    const title = screen.getByText('이달의예산');
    expect(title).toBeInTheDocument();
  });
});
