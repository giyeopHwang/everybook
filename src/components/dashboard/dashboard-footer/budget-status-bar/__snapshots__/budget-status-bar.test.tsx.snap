// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BudgetStatusBar > renders BudgetStatusBar component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1mrw3dz MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          이달의예산
        </span>
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="60"
          class="MuiLinearProgress-root MuiLinearProgress-colorSuccess MuiLinearProgress-determinate css-1d5h4r2 css-1hxt08m-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorSuccess MuiLinearProgress-bar1Determinate css-h8g9t7-MuiLinearProgress-bar1"
            style="transform: translateX(-40%);"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-1xxe9l9 css-122s7qa-MuiTypography-root"
        >
          3,000,000
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          /
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          5,000,000
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1mrw3dz MuiBox-root css-0"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        이달의예산
      </span>
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        class="MuiLinearProgress-root MuiLinearProgress-colorSuccess MuiLinearProgress-determinate css-1d5h4r2 css-1hxt08m-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorSuccess MuiLinearProgress-bar1Determinate css-h8g9t7-MuiLinearProgress-bar1"
          style="transform: translateX(-40%);"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-1xxe9l9 css-122s7qa-MuiTypography-root"
      >
        3,000,000
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        /
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        5,000,000
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`BudgetStatusBar > renders with hidden amount 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1mrw3dz MuiBox-root css-0"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          이달의예산
        </span>
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="60"
          class="MuiLinearProgress-root MuiLinearProgress-colorSuccess MuiLinearProgress-determinate css-1d5h4r2 css-1hxt08m-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorSuccess MuiLinearProgress-bar1Determinate css-h8g9t7-MuiLinearProgress-bar1"
            style="transform: translateX(-40%);"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-1xxe9l9 css-122s7qa-MuiTypography-root"
        >
          *********
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          /
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
        >
          *********
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1mrw3dz MuiBox-root css-0"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        이달의예산
      </span>
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        class="MuiLinearProgress-root MuiLinearProgress-colorSuccess MuiLinearProgress-determinate css-1d5h4r2 css-1hxt08m-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorSuccess MuiLinearProgress-bar1Determinate css-h8g9t7-MuiLinearProgress-bar1"
          style="transform: translateX(-40%);"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-1xxe9l9 css-122s7qa-MuiTypography-root"
      >
        *********
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        /
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-9o0qrs-MuiTypography-root"
      >
        *********
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
